File: Cubixcraft_v0.0.3-alpha_2021.05.18_java_src_IntelliJidea/src/game/cubixcraft/block/BlockGrass.java
package game.cubixcraft.block;

import game.cubixcraft.world.World;
import game.cubixcraft.utils.Vec2i;

public class BlockGrass extends Block {

    public BlockGrass(World world, Vec2i textureCoord) {
        super(world, BLOCK_GRASS_ID, textureCoord, "Grass");
    }

    @Override
    public Vec2i getTextureCoordinates(int sideId) {
        switch (sideId) {
            case BLOCK_SIDE_TOP:
                return new Vec2i(0,0);
            case BLOCK_SIDE_BOTTOM:
                return new Vec2i(2, 0);
            default:
                return new Vec2i(1, 0);
        }
    }
}

File: Cubixcraft_v0.0.3-alpha_2021.05.18_src_VisualC++2019/Cubixcraft/BlockGrass.cpp
#include "BlockGrass.h"
#include "world.h"

CBlockGrass::CBlockGrass(CWorld* world, POINT* textureCoord) :
	CBlock(world, BLOCK_GRASS_ID, textureCoord, L"Grass") {}

POINT* CBlockGrass::GetTextureCoordinates(int sideId)
{
	switch (sideId)
	{
		case BLOCK_SIDE_TOP:
			return new POINT{ 0, 0 };
		case BLOCK_SIDE_BOTTOM:
			return new POINT{ 2,0 };
		default:
			return new POINT{ 1,0 };
	}
}
